src/Configuration.php

```diff
@@ -1818,23 +1912,23 @@ private static function retrieveShebang(stdClass $raw): ?string
 
     private static function retrieveSigningAlgorithm(stdClass $raw): int
     {
-        // TODO: trigger warning: if no signing algorithm is given provided we are not in dev mode
-        // TODO: trigger a warning if the signing algorithm used is weak
-        // TODO: no longer accept strings & document BC break
         if (false === isset($raw->algorithm)) {
-            return Phar::SHA1;
+            return self::DEFAULT_SIGNING_ALGORITHM;
         }
 
-        if (false === defined('Phar::'.$raw->algorithm)) {
-            throw new InvalidArgumentException(
-                sprintf(
-                    'The signing algorithm "%s" is not supported.',
-                    $raw->algorithm
-                )
-            );
-        }
+        $algorithm = strtoupper($raw->algorithm);
+
+        Assertion::inArray($algorithm, array_keys(get_phar_signing_algorithms()));
 
-        return constant('Phar::'.$raw->algorithm);
+        Assertion::true(
+            defined('Phar::'.$algorithm),
+            sprintf(
+                'The signing algorithm "%s" is not supported by your current PHAR version.',
```

Comment:
wow, to be honest, I would not expect that, after we checked our algorithm is in `get_phar_signing_algorithms` :|

Reply:
Me neither, but it would be possible for example that a new PHP version introduce a new signing algorithm, in which case `get_phar_signing_algorithms ()` would have it, but the PHP version you are using does not allow it.

Just to be clear: there is no change of behaviour here in the sense it would just fail as before, the difference is the failure would happen with a more friendly error message